status.go

```diff
@@ -22,6 +22,13 @@ type HasOperatorStatus interface {
 	OperatorStatus() Status
 }
 
+// HasModificationStatus defines the comment methods for setting and getting
+// status entries for the various entities that are modified by actions.
```

Comment:
This also doesn't define what values would be allowed, and why one would be setting them or what they mean.
I'm not 100% sure where it should be, but certainly it should be clearly defined so people can understand why they should SetModificationStatus instead of SetInstanceStatus instead of SetAgentStatus.

Should we be fixing the method for SetStatus, to avoid ambiguity if we are doing this work now anyway?